test/Microsoft.AspNetCore.Mvc.FunctionalTests/BasicTests.cs

```diff
@@ -489,5 +490,22 @@ public static IEnumerable<object[]> HtmlHelperLinkGenerationData
             Assert.Equal(422, (int)response.StatusCode);
             Assert.Equal("Can't process this!", await response.Content.ReadAsStringAsync());
         }
+
+        [Fact]
+        public async Task ApplicationAssemblyPartIsListedAsFirstAssembly()
+        {
+            // Act
+            var response = await Client.GetStringAsync("Home/GetAssemblyPartData");
+            var assemblyParts = JsonConvert.DeserializeObject<IList<string>>(response);
+            var expected = new[]
+            {
+                "BasicWebSite",
+                "Microsoft.AspNetCore.Mvc.TagHelpers",
+                "Microsoft.AspNetCore.Mvc.Razor",
```

Comment:
It doesn't seem right that these show up as parts. I'd expect the default part list to not include framework assemblies.

Reply:
We manually add these as part of our configuration - https://github.com/aspnet/Mvc/blob/9f3dfd98194d94362586bd5a8c1dccec9df38a48/src/Microsoft.AspNetCore.Mvc/MvcServiceCollectionExtensions.cs#L59-L72. I wonder if we need to configure our controller discovery to skip these assemblies.

Reply:
Ah, I see, we do that because of tag helpers? That's pretty wierd. I think we do it for tag helpers as services. All of those *as services* features do wierd stuff - I'm so sad we did that.

Just ignore this for now I guess.